internal/builtinfunctions/functions_test.go

```diff
@@ -151,6 +151,12 @@ var testData = map[string]struct {
 		false,
 		"-Inf",
 	},
+	"float-to-formatted-string-positive-whole": {
+		"floatToFormattedString",
+		[]any{11.0, "f", int64(-1)},
+		false,
+		"11",
```

Comment:
Huh. Indeed I assume this meets the technical `-1` precision definition in that `ParseFloat` would return exactly 11.0, but the fact that "smallest number of digits" includes dropping the fraction somewhat surprises me. (That is, the output loses the reminder that we have an imprecise floating point value rather than a precise integer value.)

Reply:
I too wonder about the loss of (at least) the decimal point; nevertheless, in this part of the range, the result is exact and precise (eleven is one _eight_, one _two_, and one _one_, with a decimal exponent of zero), so I guess we don't need it, at least in `f` format (in `e` format, we get `1.1e+10`, so that's fine).